File: frontend/src/composables/language.js
import { ref } from 'vue'
import { createResource } from 'frappe-ui'

export const defaultLanguage = ref(localStorage.getItem('lang') || 'vi')

export const fetchLanguage = createResource({
  url: 'go1_cms.api.language.get_language',
  cache: 'Language',
  onSuccess: (data) => {
    defaultLanguage.value = data
  },
})

export const changeLanguage = createResource({
  url: 'go1_cms.api.language.change_language',
  onError(err) {
    console.log('Language not found')
  },
})

export const showLanguage = ref(false)

File: go1_cms/api/language.py
import frappe


@frappe.whitelist()
def get_language():
    return frappe.db.get_value('User', frappe.session.user, 'language') or 'en'


@frappe.whitelist()
def change_language(lang):
    language = frappe.db.get_value('Language', lang, 'name')
    if language:
        frappe.db.set_value('User', frappe.session.user, {
            'language': lang
        })
        return lang
    else:
        frappe.throw('Language not found')
